frontend/openchat-client/src/openchat.ts

```diff
@@ -4723,30 +4767,51 @@ export class OpenChat extends OpenChatAgentWorker {
         );
     }
 
-    getUserLookupForMentions(): Record<string, UserSummary> {
+    // the key might be a username or it might be a user group name
+    getUserLookupForMentions(
+        communityMembers?: Map<string, Member>,
+    ): Record<string, UserOrUserGroup> {
         if (this._userLookupForMentions === undefined) {
-            const lookup = {} as Record<string, UserSummary>;
+            const lookup = {} as Record<string, UserOrUserGroup>;
```

Comment:
This means that if a user group and a user have the same name you can only mention the user.
What if we returned an array instead? I think we always iterate these values so we don't actually need it to be a record.

Reply:
Surely it would be crazy to allow user groups and users to have the same name?

Reply:
But at which point do we prevent it? A user group could be created, then a user with the same name joins the community. I guess it won't really work with name clashes though since we support typing "@name" manually. So maybe supporting duplicate names is unrealistic and we just leave it where user groups take priority.